[16.0] [BACKPORT] account_move_tier_validation_approver from 18.0

lmignon opened the pull request.

MateuGForgeFlow committed:
[ADD] account_move_tier_validation_approver

MateuGForgeFlow committed:
[14.0][MIG] account_move_tier_validation_approver

oca-travis committed:
[UPD] Update account_move_tier_validation_approver.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

MateuGForgeFlow committed:
[13.0][IMP] account_move_tier_validation_approver: add tests

OCA-git-bot committed:
account_move_tier_validation_approver 13.0.1.0.1

olgamarcocb committed:
Added translation using Weblate (Spanish)

etobella committed:
[FIX] account_move_tier_validation_approver: From LGPL to AGPL

OCA-git-bot committed:
[UPD] README.rst

LoisRForgeFlow committed:
[FIX] manifest website links

Commit:
[UPD] Update account_move_tier_validation_approver.pot

OCA-git-bot committed:
[UPD] README.rst

bosd committed:
[MIG] account_move_tier_validation_approver: Migration to 15.0

kevinkhao committed:
[IMP] pre-commit auto fixes

kevinkhao committed:
[18.0][MIG] account_move_tier_validation_approver: Migration to 18.0

kevinkhao committed:
[IMP] account_move_tier_validation_approver: replace problematic onchange with depends readonly=False

Commit:
[UPD] Update account_move_tier_validation_approver.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-18.0/account-invoicing-18.0-account_move_tier_validation_approver
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-18-0/account-invoicing-18-0-account_move_tier_validation_approver/it/

lmignon committed:
[MIG] account_move_tier_validation_approver: Backport from 16.0

luisDIXMIT reviewed (commented):
I can't edit responsible for approval on account.move, it is always readonly. Can you check it?

nicolas-delbovier-acsone committed:
[FIX] account_move_tier_validation_approver: fix approver_id always readonly.

The readonly condition on the field `approver_id` was incorrect and therefore always evaluating to True.

For x2many fields in a domain, to check if the field has some values set you should compare with the empty list `[]` and not compare with `False`.

nicolas-delbovier-acsone committed:
[IMP] account_move_tier_validation_approver: make approver_id field always readonly for users outside of group_account_manager group

nicolas-delbovier-acsone committed:
[IMP] account_move_tier_validation_approver: allow edit approver_id

add security group to allow editing account moves approver_id

nicolas-delbovier-acsone commented:
> I can't edit responsible for approval on account.move, it is always readonly. Can you check it?

It was indeed always readonly due to an error in the readonly condition. This has been fixed now.
Users of the group "account.group_account_manager" can now edit the `approver_id` field if no reviews are already linked to the record

luisDIXMIT reviewed (approved):
Tested on runboat and LGTM!

nihelgabsi-acsone reviewed (approved):
Functional test + code review
